tools/eslint-plugin/dist/rules/no-internal-barrel-imports.js

```diff
@@ -0,0 +1,297 @@
+          const transitiveImportInfo = importedModule.resolvedModules.get(
+            importSpecifier.text
+          );
+          const reExportsExternalPackage =
+            transitiveImportInfo === undefined ||
+            transitiveImportInfo.isExternalLibraryImport;
+          return !reExportsExternalPackage;
+        });
+
+        const hasNamespaceImport =
+          importNodeTs.importClause &&
+          importNodeTs.importClause.namedBindings &&
+          importNodeTs.importClause.namedBindings.kind ===
+            ts.SyntaxKind.NamespaceImport;
+
+        // there may be some situations where not reporting this is preferable
+        const isSideEffectImport =
+          !importNodeTs.importClause ||
+          (importNodeTs.importClause.name === undefined &&
+            importNodeTs.importClause.namedBindings === undefined);
+
+        // there may be some situations where not reporting this is preferable
+        const hasDefaultImport =
+          importNodeTs.importClause &&
+          importNodeTs.importClause.name !== undefined;
+
+        if (!containsReExport) return;
+
+        const namedImports =
+          importNodeTs.importClause &&
+          importNodeTs.importClause.namedBindings &&
+          ts.isNamedImports(importNodeTs.importClause.namedBindings)
+            ? importNodeTs.importClause.namedBindings.elements
+            : [];
+
+        context.report({
+          node,
+          messageId: messageIds.noInternalBarrelImports,
+          // fixer only supports property imports
+          ...(!hasDefaultImport &&
+            !isSideEffectImport &&
+            !hasNamespaceImport &&
+            namedImports.length !== 0 && {
+              fix(fixer) {
+                return [
+                  fixer.remove(node),
+                  // first get key value pairs of a Map<ModulePaths, ImportPropsFromModule>
+                  ...Array.from(
+                    namedImports
+                      .map((namedImport) => {
+                        const symbol = checker.getSymbolAtLocation(
+                          namedImport.name
+                        );
+                        return {
+                          namedImport,
+                          importPath:
+                            symbol &&
+                            getRelativeImportPathForExportedSymbol(
+                              checker.getAliasedSymbol(symbol)
+                            ),
+                        };
+                      })
+                      .filter(({ importPath }) => importPath !== undefined)
+                      .reduce((result, cur) => {
+                        if (!result.has(cur.importPath))
+                          result.set(cur.importPath, [cur]);
+                        else result.get(cur.importPath).push(cur);
+                        return result;
+                      }, new Map())
+                  )
+                    // sort the modules (map keys)
```

Comment:
have you made sure you're not duplicating functionality of the sort-imports rule here? ESlint runs all rules a second time after you apply fixes, so your new imports should be sorted anyway. Of course, some of the current import style is covered by the import sorter VS Code extension and not an ESlint rule so you might want to replicate its functionality here.

Reply:
When running the rule with `--fix`, imports were not being sorted nor combined as I expected by the behavior you're referring to. So I implemented trivial sorting anyway to make the autofix do the work for me.
```ts
import {A, B, C} from "./Barrel";
// becomes
import {A} from "./transitive-dep-1";
import {B} from "./transitive-dep-2";
import {C} from "./transitive-dep1";
```
where the sorted order of imported properties doesn't match the order of modules imported by the barrel file.

Reply:
eh, I really need to fix this import sorting situation someday, that rule is quite useless without the VS Code extension. But I can never get around to it as it's a low priority work item. OK, let's keep it this way for now, and I'll try to remember to de-duplicate it if I ever get the sorting rules fixed.